src/lib.rs

```diff
@@ -1,7 +1,87 @@
+//! an asynchronous, Byzantine fault tolerant consensus algorithm.
+//!
+//! This crate only implements the protocol itself and is meant to be instantiated in a number of
+//! nodes known to each other by some unique identifiers (IDs), that are able to exchange
+//! authenticated (e.g. cryptographically signed) messages. The nodes provide a user-defined type
+//! of _input_ - _transactions_ - to the instance, and handle networking:
+//! Messages received from other nodes must be passed into the instance, and messages produced by
+//! the instance need to be sent to the corresponding nodes.
+//! The algorithm's output are _batches_ of transactions. The order and content of these
+//! batches is guaranteed to be the same in all the nodes, as long as enough nodes are honest and
+//! functioning.
+//!
+//! * It is _Byzantine fault tolerant_: It works correctly even if an adversary controls a number
+//! `f` of the nodes, as long as the total number `N` of nodes is greater than `3 * f`. The
+//! adversary is assumed to be unable to break cryptography, but otherwise can send any kind of
+//! message through their nodes.
+//! * It is also _asynchronous_: It does not make timing assumptions about message delivery. The
+//! adversary can even be allowed to fully control the network scheduling, i.e. arbitrarily delay
+//! messages, even those sent from an honest node to another honest node, as long as every message
+//! eventually arrives. In particular, the algorithm is independent of time scales: it works via
+//! fiber optics and via smoke signals, with the same code and the same parameters.
+//!
+//! Consensus algorithms are a fundamental building block of resilient systems that are distributed
+//! across a number `N > 3 * f` of machines, and that can tolerate any kind of failure (including
+//! complete takeover by an attacker) in up to `f` machines. Use-cases include distributed
+//! databases and blockchains.
+//!
+//! Honey Badger is modular, and composed of several other algorithms that can also be used
+//! independently. All of them are Byzantine fault tolerant and asynchronous.
+//!
+//! ## Algorithms
+//!
+//! [**Honey Badger BFT**](honey_badger/index.html)
+//!
+//! The nodes input any number of _transactions_ (any user-defined type) and outputs a sequence of
+//! _batches_. The batches have sequential numbers (_epochs_) and contain a set of transactions
+//! that were input by the nodes. The sequence and contents of the batches will be the same in all
+//! nodes.
+//!
+//! [**Common Subset**](common_subset/index.html)
+//!
+//! Each node inputs one item. The output is a set of at least `N - f` nodes' items and will be the
+//! same in every node.
+//!
+//! This is the main building block of Honey Badger: In each epoch, every node proposes a number of
+//! transactions. Using the Common Subset protocol, they agree on at least `N - f` of those
+//! proposals. The batch contains the union of these sets of transactions.
```

Comment:
In the code, the definition of output of `CommonSubset` is slightly more technical but I agree that the description of  output as something that _contains_ the union of sets of transactions is totally valid.

Reply:
You're right, strictly speaking we're not outputting a set of proposals, but a set of pairs `(proposer, proposal)`…

Maybe expanding the text (the paragraph before this one) as follows would be more accurate; or would that be unnecessarily complicated?
"The output is a set of at least `N - f` nodes' IDs, together with their proposed items, and will be the same in every node."

Reply:
The current phrasing conveys the meaning of the output mapping very well and relates it with the paper. The mapping from IDs to proposals is just our way to keep some useful identification for test purposes that would have otherwise been lost.